plotting/latex_tables/defines.tex

```diff
@@ -0,0 +1,259 @@
+\newcommand{\thmlabel}[1]   {\label{thm:#1}}
+\newcommand{\thmref}[1]         {Theorem~\ref{thm:#1}}
+\newcommand{\thmabbrevref}[1]         {Thm.~\ref{thm:#1}}
+\newcommand{\claimlabel}[1]         {\label{claim:#1}}
+\newcommand{\claimref}[1]         {Claim~\ref{claim:#1}}
+\newcommand{\thmreftwo}[2]      {Theorems \ref{thm:#1} and~\ref{thm:#2}}
+\newcommand{\lemlabel}[1]   {\label{lem:#1}}
+\newcommand{\lemref}[1]         {Lemma~\ref{lem:#1}}
+\newcommand{\algolabel}[1]   {\label{alg:#1}}
+\newcommand{\algoref}[1]         {Algorithm~\ref{alg:#1}}
+\newcommand{\lemreftwo}[2]      {Lemmas \ref{lem:#1} and~\ref{lem:#2}}
+\newcommand{\lemrefthree}[3]    {Lemmas \ref{lem:#1}, \ref{lem:#2}, and~\ref{lem:#3}}
+\newcommand{\corlabel}[1]   {\label{cor:#1}}
+\newcommand{\corref}[1]         {Corollary~\ref{cor:#1}}
+\newcommand{\nonlabel}[1]    {\label{blank:#1}}
+\newcommand{\nonref}[1]          {~(\ref{blank:#1})}
+\newcommand{\eqlabel}[1]    {\label{eq:#1}}
+\newcommand{\eqreff}[1]          {(\ref{eq:#1})}
+\renewcommand{\eqref}[1]          {Eq.~\ref{eq:#1}}
+\newcommand{\eqreftwo}[2]       {(\ref{eq:#1}) and~(\ref{eq:#2})}
+\newcommand{\ineqlabel}[1]    {\label{ineq:#1}}
+\newcommand{\ineqref}[1]        {Inequality~(\ref{ineq:#1})}
+\newcommand{\ineqreftwo}[2]     {Inequalities (\ref{ineq:#1}) and~(\ref{ineq:#2})}
+\newcommand{\invref}[1]         {Invariant~\ref{inv:#1}}
+\newcommand{\deflabel}[1]    {\label{def:#1}}
+\newcommand{\defref}[1]         {Definition~\ref{def:#1}}
+\newcommand{\propref}[1]        {Property~\ref{prop:#1}}
+\newcommand{\propreftwo}[2]     {Properties~\ref{prop:#1} and~\ref{prop:#2}}
+\newcommand{\proplabel}[1]        {\label{prop:#1}}
+
+\newcommand{\caseref}[1]        {Case~\ref{case:#1}}
+\newcommand{\casereftwo}[2]     {Cases \ref{case:#1} and~\ref{case:#2}}
+\newcommand{\lilabel}[1]        {\label{li:#1}}
+\newcommand{\liref}[1]          {line~\ref{li:#1}}
+\newcommand{\Liref}[1]          {Line~\ref{li:#1}}
+\newcommand{\lirefs}[2]         {lines \ref{li:#1}--\ref{li:#2}}
+\newcommand{\Lirefs}[2]         {Lines \ref{li:#1}--\ref{li:#2}}
+\newcommand{\lireftwo}[2]       {lines \ref{li:#1} and~\ref{li:#2}}
+\newcommand{\lirefthree}[3]     {lines \ref{li:#1}, \ref{li:#2}, and~\ref{li:#3}}
+\newcommand{\exref}[1]          {Exercise~\ref{ex:#1}}
+\newcommand{\princref}[1]       {Principle~\ref{prop:#1}}
+
+\newcommand{\obslabel}[1]   {\label{obs:#1}}
+\newcommand{\obsref}[1]         {Observation~\ref{obs:#1}}
+
+
+\newcommand{\resultref}[1]         {Result~\ref{result:#1}}
+\newcommand{\resultlabel}[1]   {\label{result:#1}}
+\newcommand{\resultreftwo}[2]      {Results~\ref{result:#1} and~\ref{result:#2}}
+\newcommand{\resultrefthree}[3]    {Results~\ref{result:#1}, \ref{result:#2}, and~\ref{result:#3}}
+\newcommand{\resultrefthrough}[2]      {Results~\ref{result:#1}-\ref{result:#2}}
+
+
+\definecolor{bg}{rgb}{0.95,0.95,0.95}
+%%% Local Variables:
+%%% mode: latex
+%%% TeX-master: "main.tex"
+%%% End:
+
+
```

Comment:
I'm not sure we need all of these defs, though if it'll be a pain to only include the necessary ones you don't have to make any changes